lib/internal/fs/rimraf.js

```diff
@@ -0,0 +1,249 @@
+// The code in this file is adapted from the rimraf module.
+// Copyright (c) Isaac Z. Schlueter and rimraf Contributors.
+
+'use strict';
+const {
+  chmod,
+  chmodSync,
+  lstat,
+  lstatSync,
+  readdir,
+  readdirSync,
+  rmdir,
+  rmdirSync,
+  stat,
+  statSync,
+  unlink,
+  unlinkSync
+} = require('fs');
+const { join } = require('path');
+const { setTimeout } = require('timers');
+const notEmptyErrorCodes = new Set(['ENOTEMPTY', 'EEXIST', 'EPERM']);
+const isWindows = process.platform === 'win32';
+const epermHandler = isWindows ? fixWinEPERM : _rmdir;
+const epermHandlerSync = isWindows ? fixWinEPERMSync : _rmdirSync;
+const numRetries = isWindows ? 100 : 1;
+
+
+function rimraf(path, options, callback) {
```

Comment:
If we're going to diverge from a vendored copy, mind adding a few comments that discuss the changes you made, and the motivation.

I also think we should add some notes to the Issue about why diverging will be an advantage:

1. can we move more functionality into libuv over time? e.g., if an OS exposes behavior similar to rimraf?
2. can we retire some of the retry logic over time.
3. as you stated, it allows us to have more consistent error handling.